src/checkers/htmlchecker.py

```diff
@@ -40,12 +41,18 @@ def get_latest(checker_data, pattern_name, html):
     except KeyError:
         return None
 
-    m = pattern.search(html)
-    if m is None:
+    assert pattern.groups == 1
+
+    m = pattern.findall(html)
+    if not m:
         log.debug("%s %s did not match", pattern_name, pattern)
         return None
+    if len(m) == 1:
+        result = m[0]
+    else:
+        result = max(m, key=LooseVersion)
```

Comment:
I'm not sure if using `distutils.version.LooseVersion` is the best way to compare versions, but at least it's already there.

Reply:
In the case of `url-pattern` the results will be URLs, not version strings, though in practice I guess LooseVersion will probably do the right thing.

Reply:
`LooseVersion` seems to handle any string just fine, but I'm not sure how reliable it would be to use it this way.